core/src/main/java/fr/sncf/osrd/cli/WorkerCommand.kt

```diff
@@ -225,6 +246,10 @@ class WorkerCommand : CliCommand {
                             "ERROR, exception received"
                                 .toByteArray() // TODO: have a valid payload for uncaught exceptions
                         status = "core_error".encodeToByteArray()
+                        // Stop worker and let another worker spawn eventually
+                        if (t is OSRDError && t.osrdErrorType == ErrorType.InfraSoftLoadingError) {
+                            throw t
```

Comment:
Why `throw` and not directly [`exitProcess`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/exit-process.html) directly since you know you want to kill the worker in this case? I'd say bubbling up an error is useful if you can't handle it right now, or if you expect it to be displayed to the end-user. But I believe in this case, you can handle it right now. What do you think?

Reply:
I'll let others (@Khoyo ?) react to this also, but here is my take.

* If we always want to exitProcess, we may do it [right away](https://github.com/OpenRailAssociation/osrd/pull/9439/files#diff-ce4c689b90b896953c59277e744de96c9fb100774f54ac3e0720c0575135864dR154) instead of throwing SoftLoadingError (so even remove that line).
* (minor) If this is a real use-case we want to maintain: for multi-infra cores (either worker or web API), not sure exitProcess is always what we want as we loose the worker for all infras.
* Opinionated: exitProcess (and early returns) in the middle of the code feels like a trap to maintain the code. Letting the worker recover anytime before the actual end of the main feels better.

Reply:
> If this is a real use-case we want to maintain

Multi infra workers are only there to ease opening core inside a debugger, and I'm not sure (?) that people *really* use multiple infras (vs only using it to not specify the worker key)

@Erashin @eckter 

Re throwing vs exitProcess(), at least in the jvm world I believe you'll at least get a stack trace with eg. the `context` of `t `, so I'd tend to err on the throwing side.

Reply:
Yeah, it's only to ease opening core inside a debugger for me.